test/e2e/knit_machineinfo_test.go

```diff
@@ -21,11 +21,9 @@ var _ = g.Describe("knit machineinfo tests", func() {
 
 	g.Context("With isolated, reserved CPUs", func() {
 		g.It("Produces the expected machineinfo output", func() {
-			g.Skip("broken until we refresh the machineinfo data")
 
 			cmdline := []string{
 				filepath.Join(binariesPath, "knit"),
-				"-P", filepath.Join(snapshotRoot, "proc"),
```

Comment:
...combined to this is not a good direction. We should have stronger isolation from the system, not weaker. And we should minimize the information we ignore from the virtual filesystem, not ignoring more.

It seems to me that the core issue here is that `knit` doesn't have a recommended way or flow to collect sysinfo.tgz, thus the content of the provided archive is easily out of sync, causing all kind issues.
We need to fix this step before, and/or document clearly (few lines in the README or in another README are fine) how the update process of `test/data` is supposed to look.

Reply:
I think is a misunderstanding here, machineinfo is not taking local files with -P: 
`➜  debug-tools git:(re_enable_machineinfo_test) ✗ grep -Rni "knitOpts.ProcFSRoot"            
pkg/knit/cmd/ghw/ghw.go:93:                     "/proc": knitOpts.ProcFSRoot,
pkg/knit/cmd/cpuaff.go:67:      ph := procs.New(knitOpts.Log, knitOpts.ProcFSRoot)
pkg/knit/cmd/cpuaff.go:88:              return fmt.Errorf("error getting process infos from %q: %v", knitOpts.ProcFSRoot, err)
pkg/knit/cmd/irqaff.go:77:      ih := irqs.New(knitOpts.Log, knitOpts.ProcFSRoot)
pkg/knit/cmd/irqaff.go:86:              return fmt.Errorf("error parsing irqs from %q: %v", knitOpts.ProcFSRoot, err)
pkg/knit/cmd/irqaff.go:116:     sh := softirqs.New(knitOpts.Log, knitOpts.ProcFSRoot)
pkg/knit/cmd/irqaff.go:120:             return fmt.Errorf("error parsing softirqs from %q: %v", knitOpts.ProcFSRoot, err)
pkg/knit/cmd/root.go:77:        root.PersistentFlags().StringVarP(&knitOpts.ProcFSRoot, "procfs", "P", "/proc", "procfs root")
pkg/knit/cmd/irqwatch.go:66:    ih := irqs.New(knitOpts.Log, knitOpts.ProcFSRoot)
`

Reply:
that is the reason why that file has been deleted and info.CPUVendorID = ""

Reply:
thing is, `knit` *should* consume the local files from the path given using `-P`. The combined action of `-P` and `-S` is supposed to fully isolate `knit` from host procfs and sysfs. The `--clean-procfs-info` is already too heavy handed, and arguably half a hack in the first place. Improving this part is out of scope from this PR; however, we can and should not making it any worse. This is why I think the current direction -getting rid of `-P` entirely- is not ideal.

Reply:
ok, understood